jvm/core/src/main/scala/org/scalatest/events/Event.scala

```diff
@@ -502,6 +528,15 @@ final case class TestSucceeded (
     import EventJsonHelper._
     s"""{ "eventType": "TestSucceeded", "ordinal": ${ordinal.runStamp}, "suiteName": ${string(suiteName)}, "suiteId": ${string(suiteId)}, "suiteClassName": ${stringOption(suiteClassName)}, "duration": ${duration.getOrElse("null")}, "testName": ${string(testName)}, "testText": ${string(testText)}, "recordedEvents" : [${recordedEvents.map(_.toJson).mkString(", ")}], "formatter": ${formatterOption(formatter)}, "location": ${locationOption(location)}, "rerunner": ${stringOption(rerunner)}, "threadName": ${string(threadName)}, "timeStamp": ${timeStamp} }""".stripMargin
   }
+
+  private[scalatest] def ensureSerializable(): Event = 
+    payload match {
+      case Some(p) if !serializeRoundtrip(p) =>
+        println(Resources.unableToSerializePayload(p.getClass().getName(), this.toString()))
+        copy(payload = None)
+
+      case _ => this
+    }
 }
```

Comment:
@cheeseng Seems like quite a bit of duplication. Maybe factor out a method that takes params and does this match?